src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -113,24 +115,16 @@ class FunSpecTest {
   }
 
   @Test fun overrideInvalidModifiers() {
-    val method = mock(ExecutableElement::class.java)
-    whenMock(method.modifiers).thenReturn(ImmutableSet.of(Modifier.FINAL))
-    val element = mock(Element::class.java)
-    whenMock(element.asType()).thenReturn(mock(DeclaredType::class.java))
-    whenMock(method.enclosingElement).thenReturn(element)
-    assertThrows<IllegalArgumentException> {
-      FunSpec.overriding(method)
-    }.hasMessageThat().isEqualTo("cannot override method with modifiers: [final]")
+    val classElement = getElement(InvalidOverrideMethods::class.java)
+    val methods = methodsIn(elements.getAllMembers(classElement))
 
-    whenMock(method.modifiers).thenReturn(ImmutableSet.of(Modifier.PRIVATE))
     assertThrows<IllegalArgumentException> {
-      FunSpec.overriding(method)
-    }.hasMessageThat().isEqualTo("cannot override method with modifiers: [private]")
+      FunSpec.overriding(findFirst(methods, "finalMethod"))
+    }.hasMessageThat().isEqualTo("cannot override method with modifiers: [public, final]")
 
-    whenMock(method.modifiers).thenReturn(ImmutableSet.of(Modifier.STATIC))
     assertThrows<IllegalArgumentException> {
-      FunSpec.overriding(method)
-    }.hasMessageThat().isEqualTo("cannot override method with modifiers: [static]")
+      FunSpec.overriding(findFirst(methods, "privateMethod"))
+    }.hasMessageThat().isEqualTo("cannot override method with modifiers: [private, final]")
```

Comment:
redundant test. (public final, private final) can remove if you prefer.

Reply:
Does `private open` work? Also what about `@JvmStatic` to get a static method?

Reply:
`private open` does not compile.
good idea on static. will update restoring that.